seisbench/models/dpp.py

```diff
@@ -223,3 +526,525 @@ def forward(self, x):
+
+            picks_group = [pick for pick in picks_in if pick.trace_id == trace_id]
+            for pick_in in picks_group:
+                output.append(obspy.Stream())
+
+                # TODO: create function stream_to_arrays() ...DONE
+                # returns picking windows (data)
+                times, data = self.stream_to_arrays(
+                    group,
+                    pick_in,
+                    strict=strict,
+                    flexible_horizontal_components=flexible_horizontal_components,
+                )
+
+                # TODO: create function _annotate_window(), instead of_annotate_point() ...DONE
+                # -> model prediction needs to be adapted to pytorch
+                #
+                # runs Monte Carlo Dropout (MCD) on picking windows, returning mc_pred array (as preds) containing results from MCD iteration
+                pred_times, pred_rates, preds = self._annotate_window(
+                    times, data, pick_in, argdict
+                )
+
+                # Write to output stream.
+                # Returns streams containing results from MCD iteration.
+                output[-1] += self._predictions_to_stream(
+                    pred_rates, pred_times, preds, pick_in, trace
+                )
+
+        return output
+
+    def _predictions_to_stream(self, pred_rates, pred_times, preds, pick_in, trace):
+        """
+        Converts a set of predictions to obspy streams
+
+        :param pred_rates: Sampling rates of the prediction arrays
+        :param pred_times: Start time of each prediction array
+        :param preds: The prediction arrays, each with shape (samples, channels)
+        :param trace: A source trace to extract trace naming from
+        :return: Obspy stream of predictions
+        """
+        output = obspy.Stream()
+        for (pred_time, pred_rate, pred) in zip(pred_times, pred_rates, preds):
+            for i in range(pred.shape[1]):
+                label = f"{pick_in.phase}"
+
+                trimmed_pred, f, _ = self._trim_nan(pred[:, i])
+                trimmed_start = pred_time + f / pred_rate
+                output.append(
+                    obspy.Trace(
+                        trimmed_pred,
+                        {
+                            "starttime": trimmed_start,
+                            "sampling_rate": pred_rate,
+                            "network": trace.stats.network,
+                            "station": trace.stats.station,
+                            "location": trace.stats.location,
+                            "channel": f"{self.__class__.__name__}_{label}",
+                        },
+                    )
+                )
+
+        return output
+
+    # runs Monte Carlo Dropout (MCD) on picking windows, returning mc_pred array (as preds) containing results from MCD iteration
+    def _annotate_window(self, times, data, pick_in, argdict):
```

Comment:
I removed `_annotate_window`.  Just for reference, the intended purpose of this function was:
"Run MCD on picking windows (previously cut out from data stream), returning a list of arrays each containing MCD samples for a predicted pick".
It may be helpful for implementing it into `_annotate_array` in `base.py`.